subtitles/ZAP Chat/ZAP_Chat_15_New_GUI.srt

```diff
@@ -0,0 +1,1527 @@
+
+302
+00:21:26,720 --> 00:21:30,240
+it's not really an interface. Yes, it gives you the request and response. But if you're a 
+
+303
+00:21:30,240 --> 00:21:37,760
+pentester, you shouldn't even be firing a browser to get the history of requests and responses, right?
+
+304
+00:21:37,760 --> 00:21:42,880
+There's a number of ways to get that. So in the concept of this PoC, it's very important to say
+
+305
+00:21:42,880 --> 00:21:51,280
+that we are after making sure that it has a smooth experience for the user and it's actually able to
+
+306
+00:21:51,680 --> 00:22:00,400
+use the ZAP APIs in a consistent way, right? In a way that gives that smooth interface view
+
+307
+00:22:00,400 --> 00:22:01,520
+to the user.
+
+308
+00:22:01,520 --> 00:22:06,080
+Yeah. And we want it to look good. We want people to enjoy using this interface.
+
+309
+00:22:06,080 --> 00:22:09,360
+So it's got to look good and work in the way they want it to.
+
+310
+00:22:10,160 --> 00:22:14,640
+So you've installed, you know, you've set up this app development environment,
+
+311
+00:22:14,640 --> 00:22:20,400
+and then you've installed this add-on. How can you actually add your own proof of concept?
+
+312
+00:22:21,680 --> 00:22:26,400
+So, what I'm going to do is just switch to the command line. Now,
+
+313
+00:22:27,360 --> 00:22:34,880
+I am in, because I've got a Mac, I'm in /user/Simon/Library/Application Support/ZAP_D.
+
+314
+00:22:34,880 --> 00:22:42,080
+Now, ZAP has a default directory that it uses and it's different on different
+
+315
+00:22:42,800 --> 00:22:48,960
+operating systems. If I was using standard ZAP, the release version of ZAP, it would just be ZAP
+
+316
+00:22:48,960 --> 00:22:50,240
+without the underscore d.
+
+317
+00:22:50,240 --> 00:22:51,200
+But
+
+318
+00:22:51,680 --> 00:22:56,840
+the underscore D and this is where ZAP puts its files and what you'll see is
+
+319
+00:22:56,840 --> 00:23:07,160
+there is a web UI PoC directory and there we have we've got the index the
+
+320
+00:23:07,160 --> 00:23:12,980
+example and you'll actually see the example is a directory. So what I'm going
+
+321
+00:23:12,980 --> 00:23:20,180
+to do is if you want say you want to create a new PoC proof of concept.. I'm
+
+322
+00:23:20,180 --> 00:23:27,160
+going to make a directory called Simon and then we go back here and go to the
+
+323
+00:23:27,160 --> 00:23:33,140
+top level and we'll see we now have a PoC called Simon and a click on there
+
+324
+00:23:33,140 --> 00:23:39,380
+it'll say 404 not found because it's just directory have a look here and
+
+325
+00:23:39,380 --> 00:23:45,080
+we see we've actually got an index.html. So we need an index.html. Well, if
+
+326
+00:23:45,080 --> 00:23:49,280
+you want to serve something from the route anyway. So, I'm just going to use
+
+327
+00:23:49,280 --> 00:23:49,820
+vi
+
+328
+00:23:50,180 --> 00:24:06,640
+and i'm just gonna use five for this.. No you basically you can use whatever editor or
```

Comment:
five → vi

Reply:
delete the first part because of dublicate

Reply:
But it's the other person talking in this sentence.

Reply:
It was wrong either way. Yiannis says: You don't have to use vi for this.